pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go

```diff
@@ -256,6 +259,26 @@ type PagerDutyLinkConfig struct {
 	Text string `json:"alt,omitempty"`
 }
 
+// DiscordConfig configures notifications via Discord.
+// See https://prometheus.io/docs/alerting/latest/configuration/#discord_config
+type DiscordConfig struct {
+	// Whether or not to notify about resolved alerts.
+	// +optional
+	SendResolved *bool `json:"sendResolved,omitempty"`
+	// The secret's key that contains the Discord webhook URL.
+	// The secret needs to be in the same namespace as the AlertmanagerConfig
+	// object and accessible by the Prometheus Operator.
+	APIURL v1.SecretKeySelector `json:"apiURL,omitempty"`
```

Comment:
```suggestion
	APIURL v1.SecretKeySelector `json:"apiURL"`
```

If you keep `omitempty`, it is still optional. I agree that it is quite confusing 😅

Reply:
I won't say confusing but it's like we have a checklist 📝 for converting optional fields to mandatory😅